hedera-node/hedera-app/src/main/java/com/hedera/node/app/Hedera.java

```diff
@@ -326,8 +326,9 @@ private void onStateInitialized(
      * Called by this class when we detect it is time to do migration.
      */
     private void onMigrate(
-            @NonNull final MerkleHederaState state, @NonNull final SerializableSemVers deserializedVersion) {
-        final var previousVersion = PbjConverter.toPbj(deserializedVersion.getServices());
+            @NonNull final MerkleHederaState state, @Nullable final SerializableSemVers deserializedVersion) {
+        final var previousVersion =
+                deserializedVersion == null ? null : PbjConverter.toPbj(deserializedVersion.getServices());
```

Comment:
Previously I made the mistake of not allowing the deserialized version to be null. But it is null during genesis.

Reply:
Might be worth putting the null check in the toPbj method, so the method is null-safe (rather than null-failing)...